Translated react v16.7 post (#156)
reactjs/pt-BR.reactjs.org: content/blog/2018-12-19-react-v-16-7.md

---

> title: "React v16.7: Não, Este Não é o Primeiro Com Hooks"

Author: Uhh entendo, eu achei um pouco confusa, mas talvez seja a minha interpretação, vou alterar, vlw @John-Luke e @halian-vilela

author: [acdlite]
---

> Nossa última versão inclui um correção crítica(bugfix) de desempenho para `React.lazy`. Embora não haja alterações na API, estamos lançando isso como uma versão minor em vez de um patch.

Reviewer: ```suggestion
Nossa última versão inclui um correção crítica (bugfix) de desempenho para `React.lazy`. Embora não haja alterações na API, estamos lançando isso como uma versão minor em vez de um patch.
```

## Por Que Este Bugfix é Minor Em Vez de Um Pacth? {#why-is-this-bugfix-a-minor-instead-of-a-patch}

> React segue [versão semântica](/docs/faq-versioning.html). Normalmente, isso significa que usamos versões patch para correções críticas(bugfixes), e minors para novos recursos (non-breaking). No entanto, reservamos a opção de liberar versões minor, mesmo que elas não incluam novos recursos. A motivação é reservar patches para mudanças que tenham uma chance muito baixa de quebrar. Os patches são o tipo mais importante de lançamento, porque às vezes eles contém correções críticas(bugfixes). Isso significa que os lançamentos de patch têm um nível mais alto de confiabilidade. É inaceitável que um patch introduza bugs adicionais, porque se as pessoas descofiarem dos patches, isso compromete nossa capacidade de corrigir bugs críticos quando eles surgem - por exemplo, para corrigir uma vulnerabilidade de segurança.

Reviewer: ```suggestion
React segue [versão semântica](/docs/faq-versioning.html). Normalmente, isso significa que usamos versões patch para correções críticas (bugfixes) e minors para novos recursos (non-breaking). No entanto, reservamos a opção de liberar versões minor, mesmo que elas não incluam novos recursos. A motivação é reservar patches para mudanças que tenham uma chance muito baixa de quebrar. Os patches são o tipo mais importante de lançamento, porque às vezes eles contém correções críticas (bugfixes). Isso significa que os lançamentos de patch têm um nível mais alto de confiabilidade. É inaceitável que um patch introduza bugs adicionais, porque se as pessoas começarão a desconfiar dos patches, isso compromete nossa capacidade de corrigir bugs críticos quando eles surgem - por exemplo, para corrigir uma vulnerabilidade de segurança.
```

> React segue [versão semântica](/docs/faq-versioning.html). Normalmente, isso significa que usamos versões patch para correções críticas(bugfixes), e minors para novos recursos (non-breaking). No entanto, reservamos a opção de liberar versões minor, mesmo que elas não incluam novos recursos. A motivação é reservar patches para mudanças que tenham uma chance muito baixa de quebrar. Os patches são o tipo mais importante de lançamento, porque às vezes eles contém correções críticas(bugfixes). Isso significa que os lançamentos de patch têm um nível mais alto de confiabilidade. É inaceitável que um patch introduza bugs adicionais, porque se as pessoas descofiarem dos patches, isso compromete nossa capacidade de corrigir bugs críticos quando eles surgem - por exemplo, para corrigir uma vulnerabilidade de segurança.

Author: ```suggestion
React segue [versionamento semântico](/docs/faq-versioning.html). Normalmente, isso significa que usamos versões patch para correções críticas (bugfixes) e minors para novos recursos (non-breaking). No entanto, reservamos a opção de liberar versões minor, mesmo que elas não incluam novos recursos. A motivação é reservar patches para mudanças que tenham uma chance muito baixa de quebrar. Os patches são o tipo mais importante de lançamento, porque às vezes eles contém correções críticas (bugfixes). Isso significa que os lançamentos de patch têm um nível mais alto de confiabilidade. É inaceitável que um patch introduza bugs adicionais, porque se as pessoas começarão a desconfiar dos patches, isso compromete nossa capacidade de corrigir bugs críticos quando eles surgem - por exemplo, para corrigir uma vulnerabilidade de segurança.
```

> React segue [versionamento semântico](/docs/faq-versioning.html). Normalmente, isso significa que usamos versões patch para correções críticas (bugfixes) e minors para novos recursos (non-breaking). No entanto, reservamos a opção de liberar versões minor, mesmo que elas não incluam novos recursos. A motivação é reservar patches para mudanças que tenham uma chance muito baixa de quebrar. Os patches são o tipo mais importante de lançamento, porque às vezes eles contém correções críticas (bugfixes). Isso significa que os lançamentos de patch têm um nível mais alto de confiabilidade. É inaceitável que um patch introduza bugs adicionais, porque se as pessoas começarão a desconfiar dos patches, isso compromete nossa capacidade de corrigir bugs críticos quando eles surgem - por exemplo, para corrigir uma vulnerabilidade de segurança.

Reviewer 2: ```suggestion
React segue [versionamento semântico](/docs/faq-versioning.html). Normalmente, isso significa que usamos versões patch para correções de bugs e minors para novos recursos (non-breaking). No entanto, reservamos a opção de liberar versões minor, mesmo que elas não incluam novos recursos. A motivação é reservar patches para mudanças que tenham uma chance muito baixa de quebrar. Os patches são o tipo mais importante de lançamento, porque às vezes eles contém correções críticas de bugs. Isso significa que os lançamentos de patch têm um padrão mais alto de confiabilidade. É inaceitável que um patch introduza bugs adicionais, porque se as pessoas começarem a desconfiar dos patches, isso compromete nossa capacidade de corrigir bugs críticos quando eles surgem - por exemplo, para corrigir uma vulnerabilidade de segurança.
```

Ajustei um pouco a questão que havia mencionado sobre a diferença entre os bugs e bugs críticos, creio que agora o sentido ficou mais coeso.

> Nós nunca pretendemos enviar bugs. React tem um ótima reputação de estabilidade, e nós pretendemos manter assim. Nós testamos completamente todas as versões do React antes de liberar. Isso inclui testes unitários, testes generativos (difusos), testes de integração e alimentação interna em dezenas de milhares de componentes. No entanto, às vezes cometemos erros. É por isso que, daqui para frente, nossa política será a de que, se um lançamento contiver alterações não triviais, usaremos a versão minor, mesmo que o comportamento externo seja o mesmo. Também vamos usar o minor ao alterar as APIs `unstable_`-pré-definidas.

Reviewer: ```suggestion
Nós nunca pretendemos enviar bugs. React tem um ótima reputação de estabilidade e nós pretendemos manter assim. Nós testamos completamente todas as versões do React antes de liberar. Isso inclui testes unitários, testes generativos (difusos), testes de integração e alimentação interna em dezenas de milhares de componentes. No entanto, às vezes cometemos erros. É por isso que, daqui para frente, nossa política será a de que, se um lançamento contiver alterações não triviais, usaremos a versão minor, mesmo que o comportamento externo seja o mesmo. Também vamos usar o minor ao alterar as APIs `unstable_`-pré-definidas.
```

> Nós nunca pretendemos enviar bugs. React tem um ótima reputação de estabilidade e nós pretendemos manter assim. Nós testamos completamente todas as versões do React antes de liberar. Isso inclui testes unitários, testes generativos (difusos), testes de integração e alimentação interna em dezenas de milhares de componentes. No entanto, às vezes cometemos erros. É por isso que, daqui para frente, nossa política será a de que, se um lançamento contiver alterações não triviais, usaremos a versão minor, mesmo que o comportamento externo seja o mesmo. Também vamos usar o minor ao alterar as APIs `unstable_`-pré-definidas.

Reviewer 2: ```suggestion
Nós nunca pretendemos enviar bugs. React tem uma merecida reputação de estabilidade e nós pretendemos mantê-la assim. Nós testamos completamente todas as versões do React antes de liberar. Isso inclui testes unitários, testes generativos (difusos), testes de integração e alimentação interna em dezenas de milhares de componentes. No entanto, às vezes cometemos erros. É por isso que, daqui para frente, nossa política será a de que, se um lançamento contiver alterações não triviais, usaremos a versão minor, mesmo que o comportamento externo seja o mesmo. Também vamos usar o minor ao alterar as APIs com prefixo `unstable_`.
```

> ## Posso Usar Hooks Ainda? {#can-i-use-hooks-yet}

Reviewer: Yet aqui tem o sentido de já.

Author: ```suggestion
## Já Posso Usar Hooks? {#can-i-use-hooks-yet}
```

Ainda não, mas em breve!

Na React Conf, dissemos que o 16.7 seria o primeiro lançamento a incluir Hooks. Isso foi um erro. Não deveríamos ter anexado um número de versão específico a um recurso não lançado. Nós vamos evitar isso no futuro.

Embora o 16.7 não inclua Hooks, por favor, não deduza nada sobre a linha do tempo do lançamento de Hooks. Nossos planos para hooks são inalterados:

- A [proposta dos Hooks](https://github.com/reactjs/rfcs/pull/68) foi aceita ([com pequenas alterações planejadas em resposta ao feedback](https://github.com/reactjs/rfcs/pull/68#issuecomment-439314884)).
- A [implementação](https://github.com/facebook/react/commit/7bee9fbdd49aa5b9365a94b0ddf6db04bc1bf51c) foi mesclada no repositório React (atrás de uma feature flag).
- Estamos atualmente na fase de testes e você pode esperar um lançamento público dentro de alguns meses.

> Ouvimos de muitas pessoas que querem começar a usar os Hooks em seus apps. Nós também não podemos esperar para enviá-los! Mas como Hooks muda a forma como escrevemos componentes React, Estamos tirando um tempo extra para acertar os detalhes. Agradecemos sua paciência enquanto preparamos este novo recurso interessante para difundir, o *use*.

Reviewer 2: ```suggestion
Ouvimos de muitas pessoas que querem começar a usar os Hooks em seus apps. Nós também mal podemos esperar para lançá-los! Mas como Hooks muda a forma como escrevemos componentes React, estamos tirando um tempo extra para acertar os detalhes. Agradecemos sua paciência enquanto preparamos este interessante novo recurso para que você, err...  *use*.
```
Hehauehuae, refazer piadas na tradução é foda, mas acho que assim a gente consegue manter o trocadilho. O que acha?

Saiba mais sobre [nosso roadmap](/blog/2018/11/27/react-16-roadmap.html) em nosso post anterior.

## Instalação {#installation}

React v16.7.0 está disponível no registro npm.

Para instalar React 16 com Yarn, execute:

```bash
yarn add react@^16.7.0 react-dom@^16.7.0
```

Para instalar React 16 com npm, execute:

```bash
npm install --save react@^16.7.0 react-dom@^16.7.0
```

Nós também fornecemos builds UMD de React via um CDN:

```html
<script crossorigin src="https://unpkg.com/react@16/umd/react.production.min.js"></script>
<script crossorigin src="https://unpkg.com/react-dom@16/umd/react-dom.production.min.js"></script>
```

Consulte a documentação para [instruções detalhadas de instalação](/docs/installation.html).

## Changelog {#changelog}

### React DOM {#react-dom}

* Correção o desempenho de `React.lazy` para um grande número de componentes lazily-loaded. ([@acdlite](http://github.com/acdlite) em [#14429](https://github.com/facebook/react/pull/14429))
* Limpar campos na desmontagem para evitar vazamentos de memória. ([@trueadm](http://github.com/trueadm) em [#14276](https://github.com/facebook/react/pull/14276))

> * Correção de bug com SSR e context ao mixar `react-dom/server@16.6` e `react@<16.6`. ([@gaearon](http://github.com/gaearon) em [#14291](https://github.com/facebook/react/pull/14291))

Reviewer 2: ```suggestion
* Correção de bug com SSR e context ao misturar `react-dom/server@16.6` e `react@<16.6`. ([@gaearon](http://github.com/gaearon) em [#14291](https://github.com/facebook/react/pull/14291))
```

> * Correção de uma regração de desempenho no modo de criação de perfil. ([@bvaughn](http://github.com/bvaughn) em [#14383](https://github.com/facebook/react/pull/14383))

Reviewer 2: ```suggestion
* Correção de uma regressão de desempenho no modo de criação de perfil (profiling). ([@bvaughn](http://github.com/bvaughn) em [#14383](https://github.com/facebook/react/pull/14383))
```

Adicionei o "profiling" em inglês aqui pois acho que "criação de perfil" embora não esteja incorreto, não remete tão diretamente ao profiling em uma primeira vista.
